homeassistant/components/hisense_connectlife/climate.py

```diff
@@ -0,0 +1,777 @@
+                self._parser = None
+        else:
+            self._parser = None
+
+        # Only OFF mode if parser not available (don't assume device supports all modes)
+        if not hasattr(self, "_attr_hvac_modes"):
+            self._attr_hvac_modes = [HVACMode.OFF]
+
+        # Get target_temp attribute
+        target_temp_attr = (
+            self._parser.attributes.get(StatusKey.TARGET_TEMP) if self._parser else None
+        )
+
+        # Parse propertyValueList to get temperature range
+        def parse_temperature_range(property_value_list):
+            ranges = []
+            for item in property_value_list.split(","):
+                item = item.strip()
+                if "~" in item:
+                    lower, upper = map(int, item.split("~"))
+                    ranges.append((lower, upper))
+            return ranges
+
+        # Get parsed temperature range
+        if target_temp_attr and target_temp_attr.value_range:
+            temperature_ranges = parse_temperature_range(target_temp_attr.value_range)
+        else:
+            _LOGGER.warning(
+                "Target temperature attribute or value range not found, using default range."
+            )
+            temperature_ranges = [(MIN_TEMP, MAX_TEMP)]
+
+        # Get temp_type attribute
+        temp_type_attr = device.status.get(StatusKey.T_TEMP_TYPE)
+
+        # Select appropriate temperature range based on temp_type
+        if temp_type_attr != "1":
+            # Use first range (Celsius)
+            if temperature_ranges:
+                min_temp, max_temp = temperature_ranges[0]
+            else:
+                min_temp = MIN_TEMP
+                max_temp = MAX_TEMP
+            self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        else:
+            # Use second range (Fahrenheit)
+            if len(temperature_ranges) > 1:
+                min_temp, max_temp = temperature_ranges[1]
+            else:
+                min_temp = MIN_TEMP
+                max_temp = MAX_TEMP
+            self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+        # Set attributes
+        self._attr_min_temp = min_temp
+        self._attr_max_temp = max_temp
+        _LOGGER.debug(
+            "Setting temperature limits %s-%s:%s:%s",
+            device_type.type_code,
+            device_type.feature_code,
+            self._attr_min_temp,
+            self._attr_max_temp,
+        )
+        if not hasattr(self, "_attr_fan_modes"):
+            self._attr_fan_modes = [
+                FAN_AUTO,
+                SFAN_ULTRA_LOW,
+                FAN_LOW,
+                FAN_MEDIUM,
+                FAN_HIGH,
+                SFAN_ULTRA_HIGH,
+            ]
+
+        if not hasattr(self, "_attr_swing_modes"):
+            self._attr_swing_modes = [SWING_OFF, SWING_VERTICAL]
+
+    def _setup_hvac_modes(self):
+        """Set up available HVAC modes based on device capabilities."""
+        if not self._parser:
```

Comment:
Can you do anything if the parser isn't there? If you can't we should have raised the appropriate exception (and of all places, this should warrant either the exception or at least a warning?)